File: src/main/java/no/nav/fo/veilarbregistrering/oppfolging/OppfolgingGateway.java
package no.nav.fo.veilarbregistrering.oppfolging;

import no.nav.fo.veilarbregistrering.bruker.Foedselsnummer;
import no.nav.fo.veilarbregistrering.registrering.bruker.AktiverBrukerResultat;
import no.nav.fo.veilarbregistrering.profilering.Innsatsgruppe;
import no.nav.fo.veilarbregistrering.besvarelse.Besvarelse;

public interface OppfolgingGateway {

    Oppfolgingsstatus hentOppfolgingsstatus(Foedselsnummer fodselsnummer);

    AktiverBrukerResultat aktiverBruker(Foedselsnummer foedselsnummer, Innsatsgruppe innsatsgruppe);

    AktiverBrukerResultat reaktiverBruker(Foedselsnummer fodselsnummer);

    void settOppfolgingSykmeldt(Foedselsnummer fodselsnummer, Besvarelse besvarelse);
}

File: src/main/kotlin/no/nav/fo/veilarbregistrering/oppfolging/OppfolgingGateway.kt
package no.nav.fo.veilarbregistrering.oppfolging

import no.nav.fo.veilarbregistrering.besvarelse.Besvarelse
import no.nav.fo.veilarbregistrering.bruker.Foedselsnummer
import no.nav.fo.veilarbregistrering.profilering.Innsatsgruppe

interface OppfolgingGateway {

    fun hentOppfolgingsstatus(fodselsnummer: Foedselsnummer): Oppfolgingsstatus
    fun kanReaktiveres(fodselsnummer: Foedselsnummer): Boolean?
    fun erUnderOppfolging(fodselsnummer: Foedselsnummer): Boolean
    fun arenaStatus(fodselsnummer: Foedselsnummer): ArenaStatus?

    fun aktiverBruker(foedselsnummer: Foedselsnummer, innsatsgruppe: Innsatsgruppe)
    fun reaktiverBruker(fodselsnummer: Foedselsnummer)
    fun aktiverSykmeldt(fodselsnummer: Foedselsnummer, besvarelse: Besvarelse)
}